CODE_STYLE.md

```diff
@@ -0,0 +1,3293 @@
+  # => '20 10'
+
+  # good
+  sprintf('%{first} %{second}', first: 20, second: 10)
+  # => '20 10'
+
+  format('%d %d', 20, 10)
+  # => '20 10'
+
+  # good
+  format('%{first} %{second}', first: 20, second: 10)
+  # => '20 10'
+  ```
+
+* <a name="array-join"></a>
+  Favor the use of `Array#join` over the fairly cryptic `Array#*` with
+<sup>[[link](#array-join)]</sup>
+  a string argument.
+
+  ```Ruby
+  # bad
+  %w(one two three) * ', '
+  # => 'one, two, three'
+
+  # good
+  %w(one two three).join(', ')
+  # => 'one, two, three'
+  ```
+
+* <a name="splat-arrays"></a>
+  Use `[*var]` or `Array()` instead of explicit `Array` check, when dealing
+  with a variable you want to treat as an Array, but you're not certain it's an
+  array.
+<sup>[[link](#splat-arrays)]</sup>
+
+  ```Ruby
+  # bad
+  paths = [paths] unless paths.is_a? Array
+  paths.each { |path| do_something(path) }
+
+  # good
+  [*paths].each { |path| do_something(path) }
+
+  # good (and a bit more readable)
+  Array(paths).each { |path| do_something(path) }
+  ```
+
+* <a name="ranges-or-between"></a>
+  Use ranges or `Comparable#between?` instead of complex comparison logic when
+  possible.
+<sup>[[link](#ranges-or-between)]</sup>
+
+  ```Ruby
+  # bad
+  do_something if x >= 1000 && x <= 2000
+
+  # good
+  do_something if (1000..2000).include?(x)
+
+  # good
+  do_something if x.between?(1000, 2000)
+  ```
+
+* <a name="predicate-methods"></a>
+  Favor the use of predicate methods to explicit comparisons with `==`.
+  Numeric comparisons are OK.
+<sup>[[link](#predicate-methods)]</sup>
+
+  ```Ruby
+  # bad
+  if x % 2 == 0
+  end
+
+  if x % 2 == 1
+  end
+
+  if x == nil
+  end
+
+  # good
+  if x.even?
+  end
+
+  if x.odd?
+  end
+
+  if x.nil?
+  end
+
+  if x.zero?
+  end
+
+  if x == 0
+  end
+  ```
+
+* <a name="no-non-nil-checks"></a>
+  Don't do explicit non-`nil` checks unless you're dealing with boolean
+  values.
+<sup>[[link](#no-non-nil-checks)]</sup>
+
+    ```ruby
+    # bad
+    do_something if !something.nil?
+    do_something if something != nil
+
+    # good
+    do_something if something
+
+    # good - dealing with a boolean
+    def value_set?
+      !@some_boolean.nil?
+    end
+    ```
+
+* <a name="no-BEGIN-blocks"></a>
+  Avoid the use of `BEGIN` blocks.
+<sup>[[link](#no-BEGIN-blocks)]</sup>
+
+* <a name="no-END-blocks"></a>
+  Do not use `END` blocks. Use `Kernel#at_exit` instead.
+<sup>[[link](#no-END-blocks)]</sup>
+
+  ```ruby
+  # bad
+  END { puts 'Goodbye!' }
+
+  # good
+  at_exit { puts 'Goodbye!' }
+  ```
+
+* <a name="no-flip-flops"></a>
+  Avoid the use of flip-flops.
+<sup>[[link](#no-flip-flops)]</sup>
+
+* <a name="prefer-next-in-loops"></a>
+Prefer `next` in loops instead of conditional blocks.
+<sup>[[link](#prefer-next-in-loops)]</sup>
+
+  ```Ruby
+  # bad
+  [0, 1, 2, 3].each do |item|
+    if item > 1
+      puts item
+    end
+  end
+
+  # good
+  [0, 1, 2, 3].each do |item|
+    next unless item > 1
```

Comment:
from experience.. I almost never, never use next. I prefer to use an iterator because I can easily factor that behavior out of the current context and reuse it.

If i see one I usually remove it.. :P